File: Top_Interview_Questions/1_TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // vector<int> ans;
        unordered_map<int, int> mem;
        for(int i = 0; i < nums.size(); ++i) {
            mem[nums[i]] = i;
        }
      
        for(int i = 0; i < nums.size(); ++i) {
            if(mem.count(target - nums[i]) && mem[target - nums[i]] != i)
                return {i, mem[target - nums[i]]};
        }

        return {};
    }
};

File: Question_Go/1_Two_Sum.go
func twoSum(nums []int, target int) []int {
    mem := make(map[int]int) //(num, index)
    for i, num1 := range nums {
        if j, ok := mem[target - num1]; ok {
            return []int{i, j}
        }
        mem[num1] = i
    }
    return nil
}